### PR TITLE
improve error handling

### PR DESCRIPTION
**Pull request by epoberezkin**

### CODE REVIEW

File: src/Simplex/Messaging/Agent.hs [hunk 1]
```diff
@@ -156,9 +161,7 @@ processCommand c@AgentClient {sndQ} st (corrId, connAlias, cmd) =
       withStore (getConn st cAlias) >>= \case
         SomeConn _ (DuplexConnection _ rq _) -> subscribe rq
         SomeConn _ (RcvConnection _ rq) -> subscribe rq
-        -- TODO possibly there should be a separate error type trying
-        -- TODO to send the message to the connection without RcvQueue
-        _ -> throwError PROHIBITED
+        _ -> throwError $ CONN SIMPLEX
```
Comment (efim-poberezkin) [inline src/Simplex/Messaging/Agent.hs:164]: I'm not sure SIMPLEX is descriptive enough.. RcvConnection is also simplex. Maybe `CONN ONLY_RCV`? Just a thought, not sure it's necessary to change

File: src/Simplex/Messaging/Agent.hs [hunk 2]
```diff
@@ -171,9 +174,7 @@ processCommand c@AgentClient {sndQ} st (corrId, connAlias, cmd) =
       withStore (getConn st connAlias) >>= \case
         SomeConn _ (DuplexConnection _ _ sq) -> sendMsg sq
         SomeConn _ (SndConnection _ sq) -> sendMsg sq
-        -- TODO possibly there should be a separate error type trying
-        -- TODO to send the message to the connection without SndQueue
-        _ -> throwError PROHIBITED -- NOT_READY ?
+        _ -> throwError $ CONN SIMPLEX
```
Comment (efim-poberezkin) [inline src/Simplex/Messaging/Agent.hs:177]: same here, but `RCV_CONN`

File: src/Simplex/Messaging/Agent/Client.hs [hunk 3]
```diff
@@ -273,4 +277,12 @@ mkAgentMessage encKey senderTs agentMessage = do
               agentMessage
             }
   paddedSize <- asks paddedMsgSize
-  liftError CRYPTO $ C.encrypt encKey paddedSize msg
+  liftError cryptoError $ C.encrypt encKey paddedSize msg
+
+cryptoError :: C.CryptoError -> AgentErrorType
+cryptoError = \case
+  C.CryptoLargeMsgError -> CMD LARGE
```
Comment (efim-poberezkin) [inline src/Simplex/Messaging/Agent/Client.hs:284]: Is it secure to differentiate this error? Maybe use the same response for all crypto related errors? Not sure again, just a thought..

**Review by efim-poberezkin — APPROVED** [review]

File: src/Simplex/Messaging/Agent/Client.hs (hunk 3 above)
Comment (efim-poberezkin) [reply src/Simplex/Messaging/Agent/Client.hs:284]: nvm I guess, I was thinking of something like padding oracle attack, but this is not similar and doesn't really leak anything not already known

File: src/Simplex/Messaging/Agent.hs [hunk 4]
```diff
@@ -242,47 +243,48 @@ processSMPTransmission c@AgentClient {sndQ} st (srv, rId, cmd) = do
...
+      sendAck c rq
       return ()
     SMP.END -> do
       removeSubscription c connAlias
       logServer "<--" c srv rId "END"
       notify connAlias END
-    _ -> logServer "<--" c srv rId $ "unexpected:" <> bshow cmd
+    _ -> do
+      logServer "<--" c srv rId $ "unexpected: " <> bshow cmd
+      notify connAlias . ERR $ BROKER UNEXPECTED
```
Comment (epoberezkin) [inline src/Simplex/Messaging/Agent.hs:287]: sendAck

**Pull request merged**